openarm_hardware/src/canbus.cpp

```diff
@@ -21,6 +23,14 @@ CANBus::CANBus(const std::string& interface) {
     addr.can_family = AF_CAN;
     addr.can_ifindex = ifr.ifr_ifindex;
 
+    if (mode_ == CAN_MODE_FD) {
+        int enable_canfd = 1;
+        if (setsockopt(sock_, SOL_CAN_RAW, CAN_RAW_FD_FRAMES, &enable_canfd, sizeof(enable_canfd)) < 0) {
+            perror("CAN FD setsockopt failed");
+            exit(EXIT_FAILURE);
```

Comment:
In general, library shouldn't call `exit()` because applications may want to control how to exit on error.